pkg/endpoint/policy.go

```diff
@@ -573,24 +574,19 @@ func (e *Endpoint) regeneratePolicy(owner Owner) error {
 	// the regeneration of the endpoint to complete.
 	policyChanged := l3PolicyChanged || l4PolicyChanged
 
-	totalRegeneration := time.Since(regenerateStart)
+	logger := e.getLogger().WithFields(logrus.Fields{
+		"policyChanged":      policyChanged,
+		"forcedRegeneration": forceRegeneration,
+	})
 
-	e.getLogger().WithFields(logrus.Fields{
-		"policyChanged":                  policyChanged,
-		"forcedRegeneration":             forceRegeneration,
-		logfields.PolicyRegenerationTime: totalRegeneration.String(),
-	}).Info("Completed endpoint policy recalculation")
+	ok, totalRegeneration := safetime.TimeSinceSafe(regenerateStart, logger)
+
+	logger.WithField(logfields.PolicyRegenerationTime, totalRegeneration.String()).
+		Info("Completed endpoint policy recalculation")
 
-	regenerateTimeSec := totalRegeneration.Seconds()
 	metrics.PolicyRegenerationCount.Inc()
-	if regenerateTimeSec < 0 {
-		e.getLogger().WithFields(logrus.Fields{
-			"policyChanged":                  policyChanged,
-			"forcedRegeneration":             forceRegeneration,
-			logfields.PolicyRegenerationTime: totalRegeneration.String(),
-			"regenerateTimeSec":              regenerateTimeSec,
-		}).Warn("BUG: regenerateTimeSec is less than 0")
-	} else {
+	if ok {
```

Comment:
Since we know the time is now safe, it should be fine to skip this ok check. If we keep it, I'm not sure that the wrapper has done anything to simplify the code, only added more of it. The same argument applies in `spanstat.go`.
The other reviewers should opine here, maybe we shouldn't track the failed cases?

Reply:
SGTM, as calling the involved functions with `time.Duration(0)` doesn't have any effect.